sim/midas/src/main/scala/midas/core/FPGATop.scala

```diff
@@ -62,13 +62,17 @@ class FPGATopIO(implicit val p: Parameters) extends WidgetIO {
 
 /** Specifies the size and width of external memory ports */
 case class HostMemChannelParams(
-  size: BigInt,
-  beatBytes: Int,
-  idBits: Int,
-  maxXferBytes: Int = 256)
+    size: BigInt,
+    beatBytes: Int,
+    idBits: Int,
+    maxXferBytes: Int = 256) {
+  def axi4BundleParams = AXI4BundleParameters(log2Ceil(size), 8 * beatBytes, idBits) 
```

Comment:
Probably best to use keyword arguments here.

Reply:
Sounds good.